src/components/Settings/Form.tsx

```diff
@@ -0,0 +1,47 @@
+import {
+  SpruceForm,
+  ValidateProps,
+  GetFormSchema,
+} from "components/SpruceForm";
+import { SettingsState } from "./Context";
+import { SettingsRoutes } from "./types";
+
+type FormProps<
+  T extends SettingsRoutes,
```

Comment:
Sorry, I didn't get a chance to review the last pr. But is the implication that you would need to add the type from the settings page to the types directory here?

Could we make it more generic/ use generics so that this component doesn't need to rely on page-specific logic? 

I ask because I would imagine in the future once we add another settings page the types would likely mix up.

Reply:
Well we definitely are using generics here :) Adding another settings section can be done by extending the `SettingsRoutes` type with e.g. `WritableDistroSettingsType`.

https://github.com/evergreen-ci/spruce/blob/e9fd310365ed7cdc2886d122b4c63492a0fef59b/src/components/Settings/types.ts#L3

I've already done most of the work to set up distro settings for EVG-19940 and since [context is set with a specific type](https://github.com/evergreen-ci/spruce/blob/e9fd310365ed7cdc2886d122b4c63492a0fef59b/src/pages/projectSettings/Context.tsx#L18) there hasn't been any type collision.

I think this is the best approach because it allows us to enforce that other parts of the context like the map of form state types include all pages associated with settings since the typing is more specific than something like `Record<string, string>`. Plus, there is some overhead required to set up a new settings context, so adding a type to `SettingsRoutes` would not be likely to be overlooked.

Reply:
Or we can just set `type SettingsRoutes = string` so that `T extends string`? Let me know if you have a preference, I think the specificity could be slightly helpful. Ultimately it doesn't matter too much though since a more specific type is always supplied.

Reply:
That makes sense thanks for the explanation!